web/src/lib/components/shared-components/notification/notification.ts

```diff
@@ -18,7 +23,14 @@ export type Notification = {
 
 type DiscardAction = { type: 'discard' };
 type NoopAction = { type: 'noop' };
-type LinkAction = { type: 'link'; target: string };
+
+export type LinkAction = {
+  type: 'link';
+  target: string;
+  newTab?: boolean;
+  button?: string;
+};
+
```

Comment:
It turns out there isn't actually any existing usage of the previous link functionality I think it was added for one scenario that has since been removed. What do you think about doing this instead?
- Remove the legacy link action entirely
- Add another option to show interface to add a button to the notification
```ts
  button?: NotificationButton // (type { text: string, onClick: () => void })
```

Reply:
Good idea!